File: Practice/AOJ/ALDS1/ALDS1_04/ALDS1_04B/alds1_04b.py
# 一瞬「TiのうちSに含まれるものの個数」かと思って、「にぶたんでTi-1を超える最小のインデックスとTiを超える最小のインデックスの差をとればいけるかな」と思ってた
# 問題もっと単純で、ただ含まれてるかどうか判断するだけだった
# もちろんこんなことしなくても[値:個数]の連想配列を用意すれば事足りるんだけどな

n = int(input())
s = list(map(int, input().split()))
q = int(input())
t = list(map(int, input().split()))


def f(arr, target):
    left, right = 0, len(arr)
    while left < right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return True
        if target < arr[mid]:
            right = mid
        else:
            left = mid + 1
    return False


count = 0
for i in t:
    count += 1 if f(s, i) else 0
print(count)

File: Practice/AOJ/ALDS1/ALDS1_04/ALDS1_04B/alds1_04b.rb
n = gets.chomp.to_i
s = gets.chomp.split.map(&:to_i)
q = gets.chomp.to_i
t = gets.chomp.split.map(&:to_i)

def func(arr, target)
  left = 0
  right = arr.size
  while left < right do
    mid = (left+right)/2
    if arr[mid] == target
      return true
    end
    if target < arr[mid]
      right = mid
    else
      left = mid + 1
    end
  end
  return false
end
cnt = 0
t.each do |i|
  cnt +=  1 if func(s, i)
end
puts cnt